### ISSUE TITLE
Support for redis > 5.0.x in pyproject.toml?

### ISSUE THREAD

**Issue by epicwhale** [python-arq/arq#458]
my codebase depends on redis ^5.0.1 and I'm unable to use the latest version of arq with it
`redis = {extras = ["hiredis"], version = "^5.0.1"}`

```
Because no versions of arq match >0.26.0,<0.27.0
 and arq (0.26.0) depends on redis[hiredis] (>=4.2.0,<5), arq (>=0.26.0,<0.27.0) requires redis[hiredis] (>=4.2.0,<5).
So, because ... depends on both redis[hiredis] (^5.0.1) and arq (^0.26.0), version solving failed.
```

Any chance the version pinning can support updates to the 5.0.x track rather than locking to 5.0.0?

Another comment on github requesting this for a slightly different reason: https://github.com/samuelcolvin/arq/issues/441#issuecomment-2069248955

**Comment by nsteinmetz** [comment]
It seems that with 0.26 release, the `<5` was added:

```
dependencies = [
    'redis[hiredis]>=4.2.0,<5',
    [...]
]
```

whereas 0.25 was:

```
dependencies = [
    'redis[hiredis]>=4.2.0',
     [...]
]
```

So I was also surprised to have a conflict this morning when I tried to upgrade from 0.25 to 0.26 !

**Comment by d-qoi** [comment]
Yeah, I'll third wanting this dependency changed! The dependency check for <5 also limits us to redis versions under 7, locking us out of the toys included in 7.2 or 7.4.

**Comment by nsteinmetz** [comment]
@d-qoi the 5 is the version of the [redis-py](https://pypi.org/project/redis/) library, not a requirement on the redis server.

You can use redis-py 5.x with Redis 7.x
